### ISSUE TITLE
Support changing titlebar color

### ISSUE THREAD

**Issue by jcatrdev** [nativefier/nativefier#1296]
**Homework**

- [ ❌ ] I looked at `nativefier --help` and https://github.com/nativefier/nativefier/blob/master/API.md
- [ ✔️ ] I searched existing issues, open & closed. Yes, my question is new.
- [ ✔️ ] I'm using the latest version available at https://github.com/nativefier/nativefier/releases

**Your question**

I want to ask if it's possible to change the titlebar color or somehow customize it?  

**Context**

 - Nativefier: Latest
 - Node.js: Latest
 - Npm: Latest
 - OS: Windows 11 21H2 (Build: 22000.258)

**Comment by daskimmel** [comment]
@ronjouch  Would you prefer a vanilla solution here or would bringing a dependency be alright? I see this project https://github.com/AlexTorresSk/custom-electron-titlebar that seems to provide exactly what the feature is looking for. I am sure we could do this without it though if you'd prefer to not have the dependency

**Comment by ronjouch** [comment]
> @ronjouch  Would you prefer a vanilla solution here or would bringing a dependency be alright? I see this project https://github.com/AlexTorresSk/custom-electron-titlebar that seems to provide exactly what the feature is looking for. I am sure we could do this without it though if you'd prefer to not have the dependency

Hi and thanks for asking.

Use your judgement 🙂. My thoughts on the matter are laid out at https://github.com/nativefier/nativefier/blob/master/HACKING.md#guidelines , point 4.

If you can do without a dep in a reasonable efficient local way, AWESOME. Else maybe a dep is legit, as long as itself good and reasonably written.

If unsure, do share your thoughts here, I or another maintainer / passerby will comment and offer perspective.

**Comment by daskimmel** [comment]
Sweet. I feel the same way as you and would prefer not to add a dep if it isn't needed. Going to play around and see if it can be done within a reasonable amount of effort without needing to bring in that other project.

**Comment by daskimmel** [comment]
@ronjouch A few of us put our heads together and were able to accomplish this by using the `title-bar-style="hidden"` flag and injecting some custom js that basically adds a custom title bar div and offers plenty of flexibility for customization by changing a few css attribute that are set within the js function. Electron provides a webkit attribute that allows to use our custom element to be able to drag the window.

This provides a solution with changing no code whatsoever, if you and the other maintainers think it's an appropriate solution.

If we did think this was valid, would this code snippet belong in CATALOG.md? It isn't for any one specific app but it is a build command contributed by the user community. We could probably add a section at the bottom that is just for "Custom Titlebar"

**Comment by ronjouch** [comment]
> If we did think this was valid, would this code snippet belong in CATALOG.md? It isn't for any one specific app but it is a build command contributed by the user community. We could probably add a section at the bottom that is just for "Custom Titlebar".

@daskimmel sure! Thanks, happy to accept a PR for that. Maybe below a separator (in markdown, `---`) and a new section making it clear these are not site-specific e.g. `General recipes`

> This provides a solution with changing no code whatsoever, if you and the other maintainers think it's an appropriate solution.
> We tested this on a few websites and on mac and windows, but still have some testing to do with linux and more websites to make sure it is sufficient.

I think that's a neat starting point. Then, if/when you get enough cross-platform experience and are ready to invest more time, you can absorb the recipes into a full-blown new feature PR.

**Comment by daskimmel** [comment]
@ronjouch so after playing around with this for a bit, we got it to work for Mac but it ended being significantly more difficult for the other OS's. The library I linked above seems to do exactly what is wanted here, but that would mean bringing in that dependency for just this cosmetic change. We could submit the injection code that makes it work for mac, or make a PR bringing in that other dependency, but only if you think its worth it.

Thanks
